src/main/java/core/basesyntax/service/FigureSupplier.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service;
+
+import core.basesyntax.enums.Color;
+import core.basesyntax.enums.FigureType;
+import core.basesyntax.figures.*;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random RANDOM = new Random();
+    private static final ColorSupplier colorSupplier = new ColorSupplier();
+    private static final double DEFAULT_RADIUS = 10;
+
+    public Figure getRandomFigure() {
+        int index = RANDOM.nextInt(FigureType.values().length);
+        FigureType figureType = FigureType.values()[index];
+        Color color = colorSupplier.getRandomColor();
+
+        return switch (figureType) {
+            case SQUARE -> new Square(color, RANDOM.nextDouble());
+            case RECTANGLE -> new Rectangle(color, RANDOM.nextDouble(), RANDOM.nextDouble());
+            case RIGHT_TRIANGLE -> new RightTriangle(color, RANDOM.nextDouble(), RANDOM.nextDouble());
+            case CIRCLE -> new Circle(color, RANDOM.nextDouble());
+            default -> new IsoscelesTrapezoid(color, RANDOM.nextDouble(), RANDOM.nextDouble(), RANDOM.nextDouble());
```

Comment:
The `RANDOM.nextDouble()` method generates a random double between 0.0 and 1.0. If you want to have a more meaningful range for the dimensions of the figures, consider scaling the random values by a factor or using a different method to generate them.